app/jobs/result_monitor_loader.rb

```diff
@@ -0,0 +1,22 @@
+# Load a result from S3 into the db
+require 'logger'
+class ResultMonitorLoader
+  @queue = :q03_pipeline_run
+  @logger = Logger.new(STDOUT)
+
+  def self.perform(pipeline_run_id, output)
+    @logger.info("Loading #{output} for pipeline run #{pipeline_run_id}")
+    pr = PipelineRun.find(pipeline_run_id)
+    return unless pr.result_status_for(output) == PipelineRun::STATUS_LOADING_QUEUED
+    begin
+      pr.update_result_status(output, PipelineRun::STATUS_LOADING)
+      pr.send(PipelineRun::LOADERS_BY_OUTPUT[output])
+      pr.update_result_status(output, PipelineRun::STATUS_LOADED)
```

Comment:
I don't quite understand how the pipeline result status is used to track the status of different loadable entities.    This looks like you are updating a status on the entire pipeline run, but all you really have information about is a specific loadable entity.

Reply:
Oh I see, `output` is a tag that describes the loadable entity.   In that case the only problem here is ensuring atomicity for these updates.